tests/diff/patch.c

```diff
@@ -610,3 +610,46 @@ void test_diff_patch__line_counts_with_eofnl(void)
 
 	git_buf_free(&content);
 }
+
+void test_diff_patch__can_strip_bad_utf8(void)
+{
+	const char *one_sha = "108e5f09fb1af8b969df1ed92e446a1e39d2c036";
+	const char *another_sha = "cdd42da2d1d53d1db2c98b7d4e972b450601c1ef";
+	git_tree *one, *another;
+	git_diff *diff;
+	git_patch *patch;
+	git_buf buf = GIT_BUF_INIT;
+	const char *expected = "diff --git a/Japanese.md b/Japanese.md\n"
+		"index dd469c9..caf022d 100644\n"
+		"--- a/Japanese.md\n"
+		"+++ b/Japanese.md\n"
+		"@@ -24,7 +24,7 @@ Redactedでは**レイヤ**における**zIndex**属性を制御可能にする \n"
+		" **zIndexはグローバルではありません!**\n"
+		" \n"
+		"-CSSとは異なり、zIndexはすべてのオブジェクトに対してグローバルではありません。zIndexプロパティは親レイヤに対してローカルです。詳細につきましては、コンテナチュートリアルで説明しています。 [TODO: Link]。\n"
+		"+CSSとは異なり、zIndexはすべてのオブジェクトに対してグローバルではありません。zIndexプロパティは親レイヤに対してローカルです。詳細につきましては、コンテナチュートリアルで説明しています。 [現在作業中です: Link]。\n"
+		" \n"
+		" ### スクリーン上のスプライトの順序付け方法\n"
+		" これまで学んだことを生かして、画面にスプライトを表示して、zIndexの設定をしてみましょう！\n";
```

Comment:
Also, we've generally tried to escape these...  the [source character set in some VS compilers is the currently running codepage](https://msdn.microsoft.com/en-us/library/mt708821.aspx).

Reply:
Ok, if you insist. I have to say https://github.com/libgit2/libgit2/pull/4542/commits/56cccc5e9babec0ab4c155013af164345a564857 looks incredibly ugly. :)